assets/js/src/common/form/select2/select2.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { SelectHTMLAttributes } from 'react';
+import classnames from 'classnames';
+import $ from 'jquery';
+import 'select2/dist/js/select2.full.js';
```

Comment:
I'm not sure if this is a good idea to build a new "fancy select" component on top of select2. Select2 is still maintained but it is quite an old solution based on jQuery. It does some magic with inline styles. Remember last time you helped me to fix some style issue in Chrome?

We already have a React wrapper of select2. See https://github.com/mailpoet/mailpoet/blob/master/assets/js/src/form/fields/selection.jsx
It is an older code but it also covers fetching data from API etc.

I mean we could rewrite it to something new but in that case, I think we should consider using some different more modern library for that. I found this one https://github.com/JedWatson/react-select but I'm sure there will be more.